jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/internal/jpa/querydef/CriteriaBuilderImpl.java

```diff
@@ -82,7 +82,7 @@ public CriteriaBuilderImpl(Metamodel metamodel){
      */
     @Override
     public CriteriaQuery<Object> createQuery(){
-        return new CriteriaQueryImpl(this.metamodel, ResultType.UNKNOWN, ClassConstants.OBJECT, this);
+        return new CriteriaQueryImpl<Object>(this.metamodel, ResultType.UNKNOWN, ClassConstants.OBJECT, this);
```

Comment:
would it make sense to turn the return type to `<T> CriteriaQuery<T>` from `CriteriaQuery<Object>` when you're here?

Reply:
@lukasj 
1) CriteriaBuilderImpl isn't a generic class, so I don't know that I see the value in such a change. True, we have `createQuery` that returns <Object> and `createTupleQuery` that returns <Tuple>. That could possibly be reduced down to just one method with generics, except for #2
2) `CriteriaBuilderImpl.createQuery()` overrides `jakarta.persistence.criteria.CriteriaBuilder.createQuery()`:
```java
public interface CriteriaBuilder {

    CriteriaQuery<Object> createQuery();

    CriteriaQuery<Tuple> createTupleQuery();
```
We couldn't make such a change without first changing the spec

Reply:
@lukasj Did I understand your point correctly? Are there any other review comments?

Reply:
it's not about CriteriaBuilderImpl being a generic class, it's about the ability to rely on/use type inference in "user" code. Since there is already similar method with the same erasure, the change does not make sense